File: CppPractice/CppLeetcode/Leetcode94.cpp
#include<vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode():val(0),left(nullptr),right(nullptr){};
    TreeNode(int x):val(x),left(nullptr),right(nullptr){};
    TreeNode(int x, TreeNode *leftp,TreeNode *rightp):val(x),left(leftp),right(rightp){};
};

class Solution {
private:
    void InOrderTraversal(TreeNode *p, vector<int> &res)
    {
        if(!p)
        {
            return;
        }
        else
        {
            InOrderTraversal(p->left,res);
            res.push_back(p->val);
            InOrderTraversal(p->right,res);
        }
    }
public:
    vector<int> inorderTraversal(TreeNode* root) 
    {
        vector<int> result;
        InOrderTraversal(root,result);
        return result;
    }
};

File: Pythonleetcode/Leetcode94.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution(object):
    def helper(self,root,res):
        if not root:
            return None

        self.helper(root.left,res)
        res.append(root.val)
        self.helper(root.right,res)


    def inorderTraversal(self, root):
        self.res=[]
        self.helper(root,self.res)
        return self.res